riscv/triggers.h

```diff
@@ -96,9 +96,15 @@ class trigger_t {
 protected:
   action_t legalize_action(reg_t val) const noexcept;
   unsigned legalize_mhselect(reg_t val, bool h_enabled) const noexcept;
+  bool textra_match(processor_t * const proc) const noexcept;
   reg_t tdata2;
 
 private:
+  mhselect_mode_t mhselect_mode(reg_t val, bool h_enabled) const noexcept {
+    val = legalize_mhselect(val, h_enabled);
+    return (mhselect_mode_t)(val == MHSELECT_MCONTEXT ?  (MHSELECT_MCONTEXT) : (val & (MHSELECT_IGNORE | MHSELECT_MCONTEXT_SELECT | MHSELECT_VMID_SELECT | MHSELECT_RESERVED)));
+  }
+  unsigned mhselect_compare() const noexcept { return (mhvalue << 1) + (mhselect >> 2); } // {mhvalue, mhselect[2]}
```

Comment:
This is not what I was suggesting, and still relies on the specific integer values of the enums (via bitwise manipulation of their values) which is not great.

What I meant was that (mhvalue=4, mhselect=4) has the exact same semantics as (mhvalue=2, mhselect=1) and therefore `mhselect_mode()` and `mhselect_compare()` should return the exact same things for those two scenarios.

Reply:
Are you suggesting to group mhselect={4,1,5} into one mode?
```
typedef enum {
  MHSELECT_IGNORE,
  MHSELECT_MCONTEXT,
  MHSELECT_VMID,
  MHSELECT_RESERVED
} mhselect_mode_t;
```
With that, the mhselect_compare() returns mhvalue or {mhvalue, mhselect[2]}.
```
unsigned mhselect_compare() { return legalize_mhselect(mhselect)==4 ? mhvalue : (mhvalue << 1) + (mhselect >> 2) }
```